src/parquet/column_writer.h

```diff
@@ -117,6 +118,10 @@ class PARQUET_EXPORT ColumnWriter {
 
   int64_t rows_written() const { return rows_written_; }
 
+  // Only considers the size of the compressed pages + page header
+  // Some values might be still buffered an not written to a page yet
+  int64_t current_compressed_bytes() const { return current_compressed_bytes_; }
```

Comment:
A couple notes

* This value is only being updated when `flush_on_close` is true, which might be surprising
* The purpose of this method from the comment is only clear with additional context about other parts of the implementation. Should this say "Total size of column chunk serialized so far" -- perhaps this should give this information regardless of whether we are flushing immediately or at the end

Reply:
Good point! I had `total_bytes_written_` in mind for the immediate flushing case.
We also keep pages in memory when the dictionary values are being encoded. Probably I will specify that the pages are only held in memory.